src/relentless/model/potential/bond.py

```diff
@@ -0,0 +1,425 @@
+    | ``r0``      | rest length.                                     |           |
+    +-------------+--------------------------------------------------+-----------+
+
+    Parameters
+    ----------
+    type : str
+        Type.
+    name : str
+        Unique name of the potential. Defaults to ``__u[id]``, where ``id`` is the
+        unique integer ID of the potential.
+
+    Attributes
+    ----------
+    coeff : :class:`BondParameters`
+        Parameters of the potential for each type.
+
+    Examples
+    --------
+    Harmonic Bond::
+
+        >>> u = relentless.potential.bond.Harmonic("A")
+        >>> u.coeff["A"].update({'k': 1.0, 'r0': 1.0})
+
+    """
+
+    def __init__(self, types, name=None):
+        super().__init__(keys=types, params=("k", "r0"), name=name)
+
+    def energy(self, type, r):
+        """Evaluate bond energy."""
+        k = self.coeff[type]["k"]
+        r0 = self.coeff[type]["r0"]
+        return k / 2 * (r - r0) ** 2
+
+    def force(self, type, r):
+        """Evaluate bond force."""
+        k = self.coeff[type]["k"]
+        r0 = self.coeff[type]["r0"]
+        return -k * (r - r0)
+
+    def derivative(self, type, var, r):
+        r"""Evaluate bond derivative with respect to a variable."""
+        k = self.coeff[type]["k"]
+        r0 = self.coeff[type]["r0"]
+        if var == "k":
+            return (r - r0) ** 2 / 2
+        elif var == "r0":
+            return -k * (r - r0)
+        else:
+            raise ValueError("Unknown parameter")
+
+
+class FENE(BondPotential):
+    r"""The Finite Extensible Nonlinear Elastic (FENE) bond interaction.
+
+    .. math::
+
+        u(r) = -\frac{k r_0^2}{2} \ln \left(1 -
+            \left(\frac{r - r_0}{r_0}\right)^2\right)
+
+    where :math:`r` is the distance between two bonded particles. The parameters
+    for each type are:
+
+    +-------------+--------------------------------------------------+-----------+
+    | Parameter   | Description                                      | Initial   |
+    +=============+==================================================+===========+
+    | ``k``       | attractive force strength.                       |           |
+    +-------------+--------------------------------------------------+-----------+
+    | ``r0``      |  size parameter.                                 |           |
+    +-------------+--------------------------------------------------+-----------+
+
+    Parameters
+    ----------
+    type : str
+        Type.
+    name : str
+        Unique name of the potential. Defaults to ``__u[id]``, where ``id`` is the
+        unique integer ID of the potential.
+
+    Attributes
+    ----------
+    coeff : :class:`BondParameters`
+        Parameters of the potential for each type.
+
+    Examples
+    --------
+    Harmonic Bond::
+
+        >>> u = relentless.potential.bond.FENE("A")
+        >>> u.coeff["A"].update({'k': 1.0, 'r0': 1.0})
```

Comment:
Same comment about type list, plus here are the typical K-G parameters for FENE:
```suggestion
    Kremer-Grest bond (no repulsion included)::

        >>> u = relentless.potential.bond.FENE("A")
        >>> u.coeff["A"].update({'k': 30, 'r0': 1.5})
```
add a second example with WCA (epsilon = 1, sigma = 1) if you add that to this class.